TestAllocatorFuzz fails from time to time with "panic: Error doing gossip broadcast: IP allocation was seeded by different peers <..>"

```bash
panic: Error doing gossip broadcast: IP allocation was seeded by different peers (received: [00:00:00:02:00:00(nick-00:00:00:02:00:00) 03:00:00:02:00:00(nick-03:00:00:02:00:00) 04:00:00:02:00:00(nick-04:00:00:02:00:00)], ours: [00:00:00:02:00:00(nick-00:00:00:02:00:00) 01:00:00:02:00:00(nick-01:00:00:02:00:00) 02:00:00:02:00:00(nick-02:00:00:02:00:00) 03:00:00:02:00:00(nick-03:00:00:02:00:00) 04:00:00:02:00:00(nick-04:00:00:02:00:00)])

goroutine 75 [running]:
github.com/weaveworks/weave/testing/gossip.(*TestRouter).run(0xc8203cab80, 0x40000020000, 0x7f4f09e7b090, 0xc8203e4500, 0xc820319b00)
	/go/src/github.com/weaveworks/weave/testing/gossip/mocks.go:126 +0x4aa
created by github.com/weaveworks/weave/testing/gossip.(*TestRouter).Connect
	/go/src/github.com/weaveworks/weave/testing/gossip/mocks.go:161 +0xa2

<..>

goroutine 81 [chan receive]:
github.com/weaveworks/weave/ipam.(*Allocator).Allocate(0xc8203e4140, 0xc82041e2a0, 0x8, 0xa000107, 0x16, 0xb35200, 0xa34100, 0x0, 0x0)
	github.com/weaveworks/weave/ipam/_test/_obj_test/allocator.go:267 +0x1f9
github.com/weaveworks/weave/ipam.TestAllocatorFuzz.func4(0xc82041e2a0, 0x8)
	/go/src/github.com/weaveworks/weave/ipam/allocator_test.go:451 +0x15e
github.com/weaveworks/weave/ipam.TestAllocatorFuzz.func9(0x320)
	/go/src/github.com/weaveworks/weave/ipam/allocator_test.go:556 +0x146
github.com/weaveworks/weave/ipam.TestAllocatorFuzz.func8.1(0xc8203f5dd0, 0xc8, 0xc8203f5dc0, 0x4)
	/go/src/github.com/weaveworks/weave/ipam/allocator_test.go:546 +0x77
created by github.com/weaveworks/weave/ipam.TestAllocatorFuzz.func8
	/go/src/github.com/weaveworks/weave/ipam/allocator_test.go:548 +0x103

FAIL	github.com/weaveworks/weave/ipam	0.517s
```

Spotted in:

* https://circleci.com/gh/weaveworks/weave/5309
* https://circleci.com/gh/weaveworks/weave/5299
* https://circleci.com/gh/weaveworks/weave/5298

--- a/ipam/allocator.go
+++ b/ipam/allocator.go
@@ -86,7 +86,7 @@ func NewAllocator(config Config) *Allocator {
 	if config.IsObserver {
 		participant = paxos.NewObserver()
 	} else {
-		participant = paxos.NewNode(config.OurName, config.OurUID, 1)
+		participant = paxos.NewNode(config.OurName, config.OurUID, 0)
 	}
 	return &Allocator{
 		ourName:     config.OurName,

--- a/ipam/allocator_test.go
+++ b/ipam/allocator_test.go
@@ -117,6 +117,7 @@ func TestBootstrap(t *testing.T) {
 
 	CheckAllExpectedMessagesSent(alloc1, alloc2)
 
+	// Check that tryPendingOps doesn't cause anything to happen here
 	alloc1.actionChan <- func() { alloc1.tryPendingOps() }
 	AssertNothingSent(t, done)
 
@@ -129,14 +130,24 @@ func TestBootstrap(t *testing.T) {
 	ExpectBroadcastMessage(alloc1, nil)
 	alloc1.OnGossipBroadcast(alloc2.ourName, alloc2.Encode())
 
-	// both nodes will get consensus now so initialize the ring
+	CheckAllExpectedMessagesSent(alloc1, alloc2)
+
+	// at this point, both nodes have seen each other, but don't have a valid proposal
+	// so there is another exchange of messages
 	ExpectBroadcastMessage(alloc2, nil)
-	ExpectBroadcastMessage(alloc2, nil)
+	alloc2.OnGossipBroadcast(alloc1.ourName, alloc1.Encode())
+
+	// we have consensus now so alloc1 will initialize the ring
+	ExpectBroadcastMessage(alloc1, nil)
+	alloc1.OnGossipBroadcast(alloc2.ourName, alloc2.Encode())
+
+	CheckAllExpectedMessagesSent(alloc1, alloc2)
+
+	// alloc2 receives the ring and does not reply
 	alloc2.OnGossipBroadcast(alloc1.ourName, alloc1.Encode())
 
 	CheckAllExpectedMessagesSent(alloc1, alloc2)
 
-	alloc1.OnGossipBroadcast(alloc2.ourName, alloc2.Encode())
 	// now alloc1 should have space
 
 	AssertSent(t, done)

--- a/ipam/paxos/paxos.go
+++ b/ipam/paxos/paxos.go
@@ -241,6 +241,9 @@ func (node *Node) pickValue() Value {
 
 // Has a consensus been reached, based on the known claims of other nodes?
 func (node *Node) Consensus() (bool, AcceptedValue) {
+	if node.quorum == 0 {
+		return false, AcceptedValue{}
+	}
 	counts := map[ProposalID]uint{}
 
 	for _, claims := range node.knows {

